doc/tutorial/ranges/index.md

```diff
@@ -0,0 +1,323 @@
+# Ranges {#tutorial_ranges}
+
+[TOC]
+
+This tutorial introduces the notion of *ranges*, a C++20 feature that SeqAn3 makes strong use of.
+
+\tutorial_head{Moderate, 60 min, \ref tutorial_concepts,}
+
+# Motivation
+
+Traditionally most generic algorithms in the C++ standard library, like std::sort, take a pair of iterators.
+If you want to sort a std::vector `v`, you have to call `std::sort(v.begin(), v.end())` and not `std::sort(v)`.
+Why was this design with iterators chosen?
+It is more flexible, because it allows e.g.:
+  * sorting only all elements after the fifth one: `std::sort(v.begin() + 5, v.end())`
+  * using non-standard iterators like reverse iterators: `std::sort(v.rbegin() + 5, v.rend())` (sorts in reverse order)
+
+But this interface is less intuitive than just calling just calling std::sort on the entity that you wish to sort and
+it allows for more mistakes, e.g. mixing two incompatible iterators.
+C++20 introduces the notion of *ranges* and provides algorithms that accept such in the namespace `std::ranges::`, e.g.
+`std::ranges::sort(v)` now works if `v` is range – and vectors are ranges!
+
+What about the two examples that suggest superiority of the iterator-based approach? In C++20 you can do the following:
+  * sorting only all elements after the fifth one: `std::ranges::sort(std::view::drop(v, 5))`
+  * sorting in reverse order: `std::ranges::sort(std::view::reverse(v))`
+
+We will discuss later what `std::view::reverse(v)` does, for now it is enough to understand that it returns something
+that appears like a container and that std::ranges::sort can sort it.
+Later we will see that this approach offers even more flexibility than working with iterators.
+
+# Ranges
+
+*Ranges* are an abstraction of "a collection of items", or "something iterable". The most basic definition
+requires only the existence of `begin()` and `end()` on the range.
+
+There are different ways to classify ranges, one way is through the capabilities of its iterator (the object
```

Comment:
I like that you explain an iterator but you have used iterator like 20 times beforehand without this explanation. So I would move this to the first mention of iterator or remove it.